pkg/source/source.go

```diff
@@ -0,0 +1,223 @@
+package source
```

Comment:
We should not limit ourselves to only GitHub urls. An SBOM could be on any URL, we should just check if this path is a url and download the file and run sbomqs on it. This is adding too much complexity.

Reply:
I hope you mean downloading the file in memory instead of local storage.

Reply:
Hey @riteshnoronha , We can't download file from any general URL in using same steps or method. For each general URL one need to construct download URL of that and it differs from server to server. For Example to download git URL , one need to construct raw content URL of it, i.e. `https://raw.githubusercontent.com`. 
Normal URL: `https://github.com/interlynk-io/sbomqs/blob/main/samples/sbomqs-spdx-syft.json`
Download URL: `https://raw.githubusercontent.com/interlynk-io/sbomqs/main/samples/sbomqs-spdx-syft.json`

Similarly, if url is google drive url, then you need to construct download URL for GDrive. 
Normal URL: `https://drive.google.com/file/d/1pL6ykrWZAIxck15tz_f2Rv2Xl3jBNSk8/view?usp=sharing`
Download URL: `https://drive.google.com/uc?export=download&id=1pL6ykrWZAIxck15tz_f2Rv2Xl3jBNSk8`

Similalry for GitLab:
Normal URL: `https://gitlab.com/viveksahu26/go-url/-/blob/main/other.spdx.txt?ref_type=heads`
Download URL: `https://gitlab.com/viveksahu26/go-url/-/raw/main/other.spdx.txt?ref_type=heads`

So, one need to know the url representing to servers.

So, it's better for now add a support specifically for git URL and later on if raise issue related to any other URLs then we can deal with that.